pinot-core/src/main/java/com/linkedin/pinot/core/operator/MProjectionOperator.java

```diff
@@ -67,15 +68,16 @@ public boolean close() {
   }
 
   @Override
-  public Block getNextBlock() {
-    Block docIdSetBlock = _docIdSetOperator.nextBlock();
-    _blockMap.put("_docIdSet", docIdSetBlock);
-    for (String column : _columnToDataSourceMap.keySet()) {
-      _blockMap.put(column, _columnToDataSourceMap.get(column).nextBlock(new BlockId(0)));
-    }
-    _currentBlock = new ProjectionBlock(_blockMap, docIdSetBlock);
-    if (_currentBlock.getDocIdSetBlock() == null) {
-      return null;
+  public ProjectionBlock getNextBlock() {
+    DocIdSetBlock docIdSetBlock = (DocIdSetBlock) _docIdSetOperator.nextBlock();
```

Comment:
are we sure we will always get DocIdSetBlock here, I thought we might get MatchEntireSegment when we dont have filters. If we are sure to get this all the time, I suggest changing the docIdSetOperator to return DocIdSetBlock and remove the cast here.